src/client/tile.cpp

```diff
@@ -1760,9 +1760,46 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 			for (u32 y = 0; y < dim.Height; y++)
 			for (u32 x = 0; x < dim.Width; x++)
 			{
-				video::SColor c = baseimg->getPixel(x,y);
+				video::SColor c = baseimg->getPixel(x, y);
 				c.setAlpha(floor((c.getAlpha() * ratio) / 255 + 0.5));
-				baseimg->setPixel(x,y,c);
+				baseimg->setPixel(x, y, c);
+			}
+		}
+		/*
+			[invert:mode
+			Inverts the given channels of the base image.
+			Mode may contain the characters "r", "g", "b", "a".
+			Only the channels that are mentioned in the mode string
+			will be inverted.
+		*/
+		else if (str_starts_with(part_of_name, "[invert:")) {
+			if (baseimg == NULL) {
+				errorstream << "generateImagePart(): baseimg == NULL "
+						<< "for part_of_name=\"" << part_of_name
+						<< "\", cancelling." << std::endl;
+				return false;
+			}
+
+			Strfnd sf(part_of_name);
+			sf.next(":");
+
+			std::string mode = sf.next("");
+			bool r = mode.find("r") != std::string::npos;
+			bool g = mode.find("g") != std::string::npos;
+			bool b = mode.find("b") != std::string::npos;
+			bool a = mode.find("a") != std::string::npos;
+
+			core::dimension2d<u32> dim = baseimg->getDimension();
+
+			for (u32 y = 0; y < dim.Height; y++)
+			for (u32 x = 0; x < dim.Width; x++)
+			{
+				video::SColor c = baseimg->getPixel(x, y);
+				if (r) c.setRed(255 - c.getRed());
+				if (g) c.setGreen(255 - c.getGreen());
+				if (b) c.setBlue(255 - c.getBlue());
+				if (a) c.setAlpha(255 - c.getAlpha());		
+				baseimg->setPixel(x, y, c);
```

Comment:
kahrl is right. Use XOR on `c.color` with the mask. `color` is a `u32` number [in the format A8R8G8B8](http://irrlicht.sourceforge.net/docu/classirr_1_1video_1_1_s_color.html#a11f2b41eeff9e7f688d322735ca2cfcf).
So you can create a mask like described above and use
```C++
c.color ^= mask;
baseimg->setPixel(x, y, c);
```